src/services/social-media.service.ts

```diff
@@ -76,50 +76,36 @@ export class SocialMediaService {
     username: string,
     publicKey: string,
   ): Promise<ExtendedPeople> {
-    let redditUser = null;
-    let foundRedditPost = null;
-
     try {
-      // Fetch data user from reddit api
-      ({data: redditUser} = await this.redditService.getActions(
+      const {data: redditUser} = await this.redditService.getActions(
         `user/${username}/about.json`,
-      ));
-
-      // Fetch post timeline based on reddit username from reddit api
-      ({data: foundRedditPost} = await this.redditService.getActions(
+      );
+      const {data: redditPost} = await this.redditService.getActions(
         `user/${username}/.json?limit=1`,
-      ));
-    } catch {
-      // ignore
-    }
-
-    if (!redditUser) throw new HttpErrors.NotFound('Invalid username');
-    if (!foundRedditPost)
-      throw new HttpErrors.NotFound('Cannot find the speccified post');
-    if (foundRedditPost.children.length === 0)
-      throw new HttpErrors.NotFound('Cannot find the spesified post');
+      );
 
-    // Verify that the publicKey is existing in user reddit
-    const foundRedditPublicKey = foundRedditPost.children[0].data.title
-      .replace(/\n/g, ' ')
-      .split(' ')
-      .find((post: string) => post === publicKey);
+      const found = redditPost.children[0].data.title
+        .replace(/\n/g, ' ')
+        .split(' ')
+        .find((post: string) => post === publicKey);
 
-    if (!foundRedditPublicKey)
-      throw new HttpErrors.NotFound('Cannot find specified post');
+      if (!found) throw new Error('PostNotFound');
 
-    return {
-      name: redditUser.subreddit.title
-        ? redditUser.subreddit.title
-        : redditUser.name,
-      originUserId: 't2_' + redditUser.id,
-      platform: PlatformType.REDDIT,
-      username: redditUser.name,
-      profilePictureURL: redditUser.icon_img
-        ? redditUser.icon_img.split('?')[0]
-        : '',
-      publicKey: publicKey,
-    } as ExtendedPeople;
+      return {
+        name: redditUser.subreddit.title
```

Comment:
please make sure if redditUser not null and then subreddit not null and the title

Reply:
how to make sure this `redditUser` is not null, as u do on `redditPost` mas @abdulhakim2902

Reply:
when fetch user from reddit, if user not exists it will automically throw an error

Reply:
got it, thanks